src/main/java/net/uaznia/lukanus/hudson/plugins/gitparameter/GitParameterDefinition.java

```diff
@@ -100,7 +103,11 @@ public GitParameterDefinition(String name, String type, String defaultValue, Str
     public ParameterValue createValue(StaplerRequest request) {
         String value[] = request.getParameterValues(getName());
         if (value == null || value.length == 0 || isBlank(value[0])) {
-            return getDefaultParameterValue();
+            if (requiredParameter) {
+                throw new Failure("Parameter: " + getName() + " is required to have a value please select an option");
```

Comment:
Are you sure the 'Failure' is good exception? I looking in Validating-String-Parameter this plugin throw AbortException, maybe it is a better solution.

Reply:
Failure works perfectly and can be thrown from within createValue